### PR TITLE
Integrate LLVM at llvm/llvm-project@6c9541d4ddfd

### PR DESCRIPTION
**Pull request by hanhanW**
Update BUILD files and submodule for
[6c9541d4ddfd](https://github.com/llvm/llvm-project/commit/6c9541d4ddfd)

### CODE REVIEW

File: llvm-bazel/llvm-project-overlay/mlir/BUILD [hunk 1]
```diff
@@ -4221,6 +4221,28 @@ cc_binary(
     deps = [
         ":IR",
         ":Support",
+        "//third_party/llvm/llvm-project/llvm:Support",
+        "//third_party/llvm/llvm-project/llvm:TableGen",
+        "//third_party/llvm/llvm-project/llvm:config",
```
Comment (GMNGeoffrey) [inline llvm-bazel/llvm-project-overlay/mlir/BUILD]: These didn't get rewritten

**Review by GMNGeoffrey — CHANGES_REQUESTED** [review]

File: llvm-bazel/llvm-project-overlay/mlir/BUILD [hunk 2]
```diff
@@ -4530,26 +4552,43 @@ gentbl(
...
 
 genlinalg(
-    name = "LinalgNamedStructuredOpsIncGen",
+    name = "LinalgNamedStructuredOpsTcIncGen",
     src = "include/mlir/Dialect/Linalg/IR/LinalgNamedStructuredOpsSpec.tc",
     linalg_outs = [
         (
-            "-gen-impl",
-            "include/mlir/Dialect/Linalg/IR/LinalgNamedStructuredOps.cpp.inc",
+            "-gen-impl -o=$@",
```
Comment (GMNGeoffrey) [inline llvm-bazel/llvm-project-overlay/mlir/BUILD:4559]: Looks like there were also changes to linalggen.bzl to not include adding the -o itself

**Comment by hanhanW** [comment]
It looks like I have to integrate a later version. How do I find a good revision?

**Comment by GMNGeoffrey** [comment]
[hit the wrong button]

File: llvm-bazel/llvm-project-overlay/mlir/BUILD (hunk 2 above)
Comment (GMNGeoffrey) [reply llvm-bazel/llvm-project-overlay/mlir/BUILD:4559]: Urgh and I'll have to investigate more deeply what's going on here...

**Comment by hanhanW** [comment]
I still hit failures:

```
ERROR: /var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/a68af9d2db60191cc582220b74f8f8ae/external/llvm-project/mlir/BUILD:4554:10: declared output 'external/llvm-project/mlir/include/mlir/Dialect/Linalg/IR/LinalgNamedStructuredOps.tcgen.cpp.inc' was not created by genrule. This is probably because the genrule actually didn't create this output, or because the output was a directory and the genrule was run remotely (note that only the contents of declared file outputs are copied from genrules run remotely)
--
  | ERROR: /var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/a68af9d2db60191cc582220b74f8f8ae/external/llvm-project/mlir/BUILD:4554:10: not all outputs were created or valid

```

**Pull request closed without merging**